acme/acme/client.py

```diff
@@ -121,18 +123,22 @@ def new_order(self, csr_pem: bytes) -> messages.OrderResource:
         :returns: The newly created order.
         :rtype: OrderResource
         """
-        csr = OpenSSL.crypto.load_certificate_request(OpenSSL.crypto.FILETYPE_PEM, csr_pem)
+        csr = x509.load_pem_x509_csr(csr_pem)
         # pylint: disable=protected-access
-        dnsNames = crypto_util._pyopenssl_cert_or_req_all_names(csr)
-        ipNames = crypto_util._pyopenssl_cert_or_req_san_ip(csr)
```

Comment:
This was the only call to `_pyopenssl_cert_or_req_san_ip`, so we can now also delete it, its tests, `_pyopenssl_extract_san_list_raw`, and bonus `_pyopenssl_cert_or_req_san_ip` which wasn't being used anyway. Doesn't need to happen here if you'd prefer.

Reply:
If you don't want to wholesale delete them, could get that 100% acme test coverage by sticking that try/except/else block in a crypto util function and modifying the no-longer-needed `PyOpenSSLCertOrReqSANIPTest` to use those existing testcases.

Reply:
Unfortunately I'm not sure we can delete it -- the way certbot specifies its dependency on acme is `acme>={version}`, which means its valid to use an old certbot with a new acme.

Thus, if we removed it, it'd be a breaking change.

Reply:
Why would it break? Are we using it somewhere in a version of certbot before this PR that I'm missing?

Reply:
I'm not sure I understand, old versions of certbot use `_pyopenssl_cert_or_req_san_ip`, so we can't delete it from acme, because its valid to use old certbot with new acme.

Reply:
That's what I'm missing  -- it looks like that function is only called from within acme, not in certbot.

```
~/certbot$ git checkout v3.0.1
HEAD is now at 93294fc98 Release 3.0.1

~/certbot$ git grep _pyopenssl_cert_or_req_san_ip
acme/acme/_internal/tests/crypto_util_test.py:    """Test for acme.crypto_util._pyopenssl_cert_or_req_san_ip."""
acme/acme/_internal/tests/crypto_util_test.py:        from acme.crypto_util import _pyopenssl_cert_or_req_san_ip
acme/acme/_internal/tests/crypto_util_test.py:        return _pyopenssl_cert_or_req_san_ip(loader(name))
acme/acme/client.py:        ipNames = crypto_util._pyopenssl_cert_or_req_san_ip(csr)
acme/acme/crypto_util.py:def _pyopenssl_cert_or_req_san_ip(cert_or_req: Union[crypto.X509, crypto.X509Req]) -> List[str]:
```

Reply:
You are quite right, and I am quite wrong! Sorry about that! Do you mind if we do the deletions in a follow-up PR?

Reply:
Didn't see this before I merged! So, yes, absolutely.